frontends/common/resolveReferences/referenceMap.h

```diff
@@ -29,28 +29,59 @@ class NameGenerator {
     virtual cstring newName(cstring base) = 0;
 };
 
+
+/// Class used to encode maps from paths to declarations.
 class ReferenceMap final : public ProgramMap, public NameGenerator {
-    bool isv1;  // if true this is a map for a P4 v1.0 program (P4-14)
-    // Maps each path in the program to the corresponding declaration
+    /// If `isv1` is true, then map is for a P4_14 program
```

Comment:
Maps for v1 programs are different because v1 programs allow declarations to come after uses.
This difference is carried even in P4-16 program that are generated from P4-14 programs.